lib/AST/AutoDiff.cpp

```diff
@@ -99,11 +98,56 @@ void autodiff::getSubsetParameterTypes(IndexSubset *subset,
     unsigned parameterIndexOffset =
         curryLevelParameterIndexOffsets[curryLevelIndex];
     for (unsigned paramIndex : range(curryLevel->getNumParams()))
-      if (subset->contains(parameterIndexOffset + paramIndex))
-        results.push_back(curryLevel->getParams()[paramIndex].getOldType());
+      if (parameterIndices->contains(parameterIndexOffset + paramIndex))
+        results.push_back(curryLevel->getParams()[paramIndex]);
   }
 }
 
+AutoDiffSemanticFunctionResultType
+autodiff::getOriginalFunctionSemanticResultType(
```

Comment:
What's the word "original" trying to disambiguate?  Seems redundant to me.

Reply:
> What's the word "original" trying to disambiguate? Seems redundant to me.

Thanks for the catch! I renamed `getOriginalFunctionSemanticResultType` to `getFunctionSemanticResultType` and simplified it.